pkg/release/release.go

```diff
@@ -35,13 +37,54 @@ func LocalVersionMetadataConfigmap(version string) types.NamespacedName {
 	}
 }
 
+// configureRegistryTLS makes sure that the docker-registry values contains an entry for the
+// tls secret. this function should be called only if the tls secret exists.
+func configureRegistryTLS(meta *ectypes.ReleaseMetadata) (*ectypes.ReleaseMetadata, error) {
+	for i, chart := range meta.AirgapConfigs.Charts {
+		if chart.Name != "docker-registry" {
+			continue
+		}
+		var values map[string]interface{}
+		if err := yaml.Unmarshal([]byte(chart.Values), &values); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal registry chart values: %w", err)
+		}
+		values["tlsSecretName"] = "registry-tls"
+		newValues, err := yaml.Marshal(values)
+		if err != nil {
+			return nil, fmt.Errorf("unable to marshal new registry chart values: %w", err)
+		}
+		meta.AirgapConfigs.Charts[i].Values = string(newValues)
+		return meta, nil
+	}
+	return meta, nil
+}
+
 // MetadataFor determines from where to read the metadata (from the cluster or remotely) and calls
-// the appropriate function.
+// the appropriate function. If running on airgap environment this function also assess if the
+// registry requires tls or not and customize the release metadata accordingly.
 func MetadataFor(ctx context.Context, in *v1beta1.Installation, cli client.Client) (*ectypes.ReleaseMetadata, error) {
-	if in.Spec.AirGap {
-		return localMetadataFor(ctx, cli, in.Spec.Config.Version)
+	if !in.Spec.AirGap {
+		return remoteMetadataFor(ctx, in.Spec.Config.Version, in.Spec.MetricsBaseURL)
+	}
+
+	meta, err := localMetadataFor(ctx, cli, in.Spec.Config.Version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get local metadata: %w", err)
+	}
+
+	var secret corev1.Secret
+	nsn := types.NamespacedName{Namespace: "registry", Name: "registry-tls"}
+	if err = cli.Get(ctx, nsn, &secret); err != nil {
+		if errors.IsNotFound(err) {
+			return meta, nil
+		}
+		return nil, fmt.Errorf("failed to get registry tls secret: %w", err)
+	}
+
+	if meta, err = configureRegistryTLS(meta); err != nil {
+		return nil, fmt.Errorf("failed to configure registry tls: %w", err)
```

Comment:
can you move this code into localMetadataFor, as it's only used for airgap?

I think that would be clearer, given that it's logically equivalent (it already only runs in airgap installs since remoteMetadataFor returns directly)

Reply:
Also, can we get a test case for localMetadataFor that has this secret present?